### ISSUE TITLE
Feature Request: Detect unintended tuple creation from trailing commas in boolean expressions

### ISSUE THREAD

**Issue by l1n** [astral-sh/ruff#14028]
While Ruff currently detects trailing commas that create unintended tuples in direct `if` statement conditions, it doesn't detect similar issues when the tuple is created in a variable assignment and then used in control flow. This can lead to subtle bugs where boolean expressions become truthy tuples instead of boolean values.

## Description
Consider this code pattern:

```python
# Currently detected by Ruff
if (condition1 or condition2(),):  # Ruff warns about this
    ...

# But not detected
should_run_tests = (
    condition1() or condition2(),  # No warning
)
if should_run_tests:  # Bug: should_run_tests is always truthy (it's a tuple)
    ...
```

The trailing comma creates a single-element tuple, which is always truthy in Python boolean contexts. This is particularly problematic because:

1. The code appears to be a standard boolean expression at first glance
2. The pattern is common in multi-line boolean expressions for readability (which is how this manifested in our codebase...)
3. The bug is subtle and may not be caught by tests (tuple is always truthy)
4. Type checkers like PyRight only catch this with explicit type annotations (which don't really cover the if True footgun)
5. The pattern is particularly dangerous in control flow contexts where boolean evaluation is expected

## Current Behavior
- Ruff detects unintended tuple creation in direct `if` statement conditions
- Does not detect similar issues in variable assignments

## Proposed FR
Extend the existing rule, [F634](https://docs.astral.sh/ruff/rules/if-tuple/) (or create a new one) to:

1. Detect trailing commas in parenthesized boolean expressions involving:
   - `and`, `or` operators
   - Function calls that typically return booleans (e.g., methods named `is_*`, `has_*`, `should_*`)
   - Boolean constants (`True`, `False`)

2. Consider context clues that suggest boolean intent:
   - Variable names with boolean prefixes (`is_*`, `has_*`, `should_*`)
   - Usage in control flow contexts (`if`, `while`, etc.)
   - Type annotations (`: bool`)

## Example Cases

```python
# Should trigger warnings:
should_run_tests = (
    is_devbox() or is_local(),  # Trailing comma creates tuple
)

has_permission = (
    check_admin() or
    check_owner() or
    check_write_access(),  # Trailing comma
)

# Should not trigger warnings:
valid_tuple = (
    1,
)

config_values = (
    get_config("a"),
    get_config("b"),
)
```

## Example Error Message
```
error: Trailing comma creates a tuple in boolean expression context, which is always truthy
   │
12 │     should_run_tests = (condition1() or condition2(),)
   │                                                     ^ trailing comma creates tuple
   │
   = help: Remove the trailing comma to create a boolean expression
   = note: Single-element tuples are always truthy in boolean contexts
```

**Comment by l1n** [comment]
Hi @MichaReiser! Sorry to bump, but I was curious whether this seemed feasible to support in F634 or something like that.

**Comment by MichaReiser** [comment]
I don't think this should be handled by F634. That rule is explicitly scoped around if-conditions,
